pkg/controller/deployment/sync.go

```diff
@@ -510,15 +510,26 @@ func (dc *DeploymentController) updateDeploymentStatus(allRSs []*extensions.Repl
 
 // isScalingEvent checks whether the provided deployment has been updated with a scaling event
 // by looking at the desired-replicas annotation in the active replica sets of the deployment.
-func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) (bool, error) {
 	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
 	if err != nil {
-		return false
+		return false, err
 	}
 	// If there is no new replica set matching this deployment and the deployment isn't paused
 	// then there is a new rollout that waits to happen
 	if newRS == nil && !d.Spec.Paused {
```

Comment:
If there is a template change then we need to rollout a new replica set. I don't think we want to do anything (scaling included) with the old replica set beforehand. Simple example: I have a router which uses host ports and the deployment's size equals the nodes in the cluster. I change my router not to use host ports anymore and I scale it up at the same time. If we try to scale before we rollout the new template then we simply won't be able to do so. Eventually I think we will rollout successfully but running scaling before rolling out is not optimal.